metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -414,7 +415,7 @@ def parse_restli_response(response):
 
 
 @ingest.command()
-@click.argument("path", type=click.Path(exists=True))
```

Comment:
the file source already supports things like s3

https://github.com/datahub-project/datahub/blob/84c677629d78a0644f667c3e0addac9fb28154ce/metadata-ingestion/src/datahub/ingestion/source/file.py#L324

let's reuse that stuff instead of adding something new

Reply:
in fact - it might actually already work out of the box, with no changes required

Reply:
To be clear, afaik that code is exactly what this PR enables to be used. This didn't work because of the `click` annotation rejecting it. An alternative solution would be to remove `click.Path` and make it a generic `string`. This solution preserves the preconditions applied by `click` but it passes the actual handling of the s3 path to that code for ingestion purposes. `click` is just getting in the way with its default implementation of `click.Path`

Reply:
Another possible alternative is to set `exists=True` to `exists=False` to prevent `click` from rejecting s3 paths.

Reply:
Oh I see - yeah let’s just remove the exists check or change it to a string. 

The file source will validate the path when reading it anyway, so no need to redo it here